test/acceptance/start.test.js

```diff
@@ -0,0 +1,37 @@
+import expect from 'expect'
+import { when } from 'interactor.js'
+
+import {
+  run,
+  setupApiProxy,
+} from './helpers'
+
+describe('percy start', () => {
+  let proxy = setupApiProxy()
+
+  it('warns and exits when PERCY_TOKEN is missing', async () => {
+    let [stdout, stderr, code] = await run('percy start', { PERCY_TOKEN: '' })
+    expect(stderr).toHaveEntry('Warning: Skipping visual tests. PERCY_TOKEN was not provided.')
+    expect(stdout).toHaveLength(0)
+    expect(code).toBe(0)
+  })
+
+  it('starts the agent server and stops it when the process is killed', async () => {
+    let start = run('percy start')
+    // kill the process 500ms after it creates a build
+    await when(() => !!proxy.requests['/builds'].length, 3000)
```

Comment:
Looks like this is failing in CI 👀

Reply:
Not so much that this was failing, but that the following kill was failing because agent wasn't really running yet. Since kill failed, the `await start` made the test timeout.

`when` was too fast and even though the build was created, the server hadn't been started. Fixed by using a more determinstic helper, but still might flake because of puppeteer launching Chrome (even though this test doesn't send snapshots, we launch chrome on server init).

Reply:
Looks like it's still failing to kill agent 🤔